extension/lib/updateProductsInCart.js

```diff
@@ -0,0 +1,26 @@
+const ShopgateCartPipeline = require('./shopgate/CartExtensionPipeline')
+
+/**
+ * @param {PipelineContext} context
+ * @param {UpdateProductsInput} input
+ * @returns {Promise<UpdateCartMessage[]>}
+ */
+module.exports = async (context, input) => {
+  try {
+    const warningsPresent = await ShopgateCartPipeline.create(context).updateProducts(input.CartItem)
+    // todo see about proper way to send an warning in case there are non breaking errors present.
+    const messages = []
+    if (warningsPresent) {
+      messages.push({
+        code: 'EUNKNOWN',
+        message: 'It was not possible to update all of the items.',
+        type: 'error'
+      })
+    }
+
+    return messages
```

Comment:
The response here is wrong it should be `return { messages }`

Reply:
corrected the return. Tested to ensure that the message's message is shown to the shopper. It was all is good.